cpp/terrain/terrain.cpp

```diff
@@ -482,11 +469,13 @@ struct tile_draw_data Terrain::create_tile_advice(coord::tile position) {
 		struct neighbor_tile neigh_data[8];
 
 		// get all neighbor tiles around position, reset the influence directions.
-		this->get_neighbors(position, neigh_data, this->influences_buf);
+		this->get_neighbors(position, neigh_data, this->influences_buf.get());
```

Comment:
get_neighbors should return a std::vector<struct neighbor_tile> instead of writing the result to a preallocated array.